nixos/modules/programs/git.nix

```
@@ -49,7 +49,16 @@ in
     (mkIf cfg.enable {
       environment.systemPackages = [ cfg.package ];
       environment.etc.gitconfig = mkIf (cfg.config != {}) {
-        text = generators.toGitINI cfg.config;
+        text =
+        let
+          flatGit = f: flatten (mapAttrsToList (type: s: mapAttrsToList (key: value: { "${type}"."${key}" = (removeAttrs value [ "_content" ]); } // (if value ? "_content" then value._content else {})) s) f);
+        in
+          (
+            generators.toGitINI (filterAttrs (type: _: type != "include" && type != "includeIf") cfg.config)
+            ) + "\n" +
+          (
+            concatMapStringsSep "\n" generators.toGitINI (flatGit (filterAttrs (type: _: type == "include" || type == "includeIf") cfg.config))
+            );
```

```suggestion
          (generators.toGitINI (filterAttrs (type: _: type != "include" && type != "includeIf") cfg.config)) + "\n" +
          (concatMapStringsSep "\n" generators.toGitINI (flatGit (filterAttrs (type: _: type == "include" || type == "includeIf") cfg.config)));
```
or
```suggestion
          (
            generators.toGitINI (filterAttrs (type: _: type != "include" && type != "includeIf") cfg.config)
          ) + "\n" +
          (
            concatMapStringsSep "\n" generators.toGitINI (flatGit (filterAttrs (type: _: type == "include" || type == "includeIf") cfg.config))
          );
```

also I think we need a comment or more here otherwise the next person will have a hard time understanding this